### ISSUE TITLE
Emulator killers

### ISSUE THREAD

**Issue by tomcw** [AppleWin/AppleWin#728]
A few from @peterferrie on Slack #coding (11th Oct 2019):

### Emulator killer:
```
800:E0 00
802<801.BFEM
80B:A2 33 A0 44 E8 C8 60 60
300:A2 11 A0 22 AD 55 C0 AD 55 C0 C9 E0 D0 F9 20 50 C0 2C 54 C0 00
```
>On hardware, I get a break at $316, and X=#$34, Y=#$44.  In AppleWin, it breaks at $C050, in MAME, it breaks at $C052.
>I'm running code from inside the soft-switch space.  No-one needs to support that.

Actually I get a break at $316, and LORES (full-screen) enabled:
0316-    A=E0 X=11 Y=22 P=B1 S=..

---

### Another one:
```
300:
    lda    $c010
-   lda    $c000
    bpl    -
    jsr    $c000
    brk
Then press back-tick. (IE. keycode=$60)

300:AD 10 C0 AD 0 C0 10 FB 20 0 C0 0
```
This works in AppleWin, with it executing CPX #imm (ie. 0xE0 nn) when in $C00n space, and executing RTS when at $C010.

The results is:
030D-    A=E0 X=...

**Comment by tomcw** [comment]
I'm recording these here, but will close as WON'T FIX.

**Comment by tomcw** [comment]
From Slack (16th Dec 2021), "[Thomas] Harte's 6502 tests"

```
Initial:    PC=017b A=9E X=89 Y=34 S=7D P=E6 {NV-B-IZ-} SP={$7D stack:[ FF=00 FE=00 FD=00 FC=00 FB=00 FA=00 F9=00 F8=00 ]}}
Final:      PC=0155 A=9E X=89 Y=34 S=7B P=E6 {NV-B-IZ-} SP={$7B stack:[ FF=00 FE=00 FD=00 FC=00 FB=00 FA=00 F9=00 F8=00 ]}}
mem[017b]=20
mem[017c]=55
mem[017d]=13
mem[0155]=ad
017B: 20 55 13   JSR $1355

Incorrect PC value, expected 155 but got 1355
```

AppleWin's opcode emulation does this:
```
	case 0x20: ABS        JSR  CYC(6)  break;
```

1) "ABS": Fetch absolute 16-bit address to JSR to ($1355)
2) "JSR": Push current PC+2

As with the other cases above, I've no interest in fixing this.

**Comment by tomcw** [comment]
Update: for this last one, I fixed the CPU emulation of JSR in #1257.
